UI Changes for Configuring Proxy-Cache Speed Limit

kunal-511 opened the pull request:
Thank you for contributing to Harbor!

proxy-cache bandwidth limit UI change
# Comprehensive Summary of your change
This pull request addresses the UI changes required for configuring the proxy-cache speed limit, as outlined in Issue #20589. The changes include the addition of a bandwidth field for setting the speed limit in both the proxy-cache creation page and the project configuration page
# Issue being fixed
Fixes #20811 

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

Vad1mo commented on src/portal/src/i18n/lang/en-us-lang.json:
```diff
@@ -246,6 +246,7 @@
         "TOGGLED_SUCCESS": "Toggled project successfully.",
         "FAILED_TO_DELETE_PROJECT": "Project contains repositories or replication rules or helm-charts cannot be deleted.",
         "INLINE_HELP_PUBLIC": "When a project is set to public, anyone has read permission to the repositories under this project, and the user does not need to run \"docker login\" before pulling images under this project.",
+        "PROXY_CACHE_BANDWIDTH":"Set the maximum network bandwidth to pull image from upstream for proxy-cache. For unlimited bandwidth, please enter -1. ",
```
this string needs to be added to all language files. (Put it in the english version), for later translation.

 Its than easier to pick it up by others and translate

kunal-511 replied:
Added this line to all the language files

stonezdj commented:
I verify the UI, the payload is wrong:
```
{"project_name":"dockerhub5","metadata":{"public":"true"},"storage_limit":-1,"registry_id":1}
```
It should be in this format
```
{"project_name":"dockerhub5","metadata":{"public":"true", "storage_limit":-1},"registry_id":1}
```

stonezdj commented:
![Screenshot 2024-08-30 at 15 00 49](https://github.com/user-attachments/assets/a8fd27a2-12ea-464e-8757-bb1914d67291)
The color in the warning message is different from the existing color.

stonezdj commented:
@kunal-511  could please refer to this PR https://github.com/stonezdj/harbor/commit/e39f73edd617afb7a933a2c4a77847d792964333 to update your code.

kunal-511 commented:
@stonezdj  Updated my PR

chlins commented:
@kunal-511 Please resolve the code conflicts.

kunal-511 commented:
> @kunal-511 Please resolve the code conflicts.

Resolved the code conflicts

kunal-511 commented:
> @kunal-511 could please refer to this PR [stonezdj@e39f73e](https://github.com/stonezdj/harbor/commit/e39f73edd617afb7a933a2c4a77847d792964333) to update your code.

need to update other language json files ??

stonezdj commented:
> @stonezdj Updated my PR

Please update the language json file to add this line.
```
        "BANDWIDTH": "Bandwidth",
        "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
```

kunal-511 commented:
> > @stonezdj Updated my PR
> 
> Please update the language json file to add this line.
> 
> ```
>         "BANDWIDTH": "Bandwidth",
>         "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
> ```

Need to add this to all language json files?

stonezdj commented:
> > > @stonezdj Updated my PR
> > 
> > 
> > Please update the language json file to add this line.
> > ```
> >         "BANDWIDTH": "Bandwidth",
> >         "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
> > ```
> 
> Need to add this to all language json files?

Yes

stonezdj commented on src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.scss:
```diff
@@ -1,6 +1,9 @@
 #severity-blk div {
   display: inline-block;
 }
+.mr-10 {
```
The current UI change is only limit to create project, why change the CSS of the project policy config?

kunal-511 replied:
> The current UI change is only limit to create project, why change the CSS of the project policy config?

It was initially told by @zyyw that we need to make changes to both the create project and also to the  project policy-config page

stonezdj commented:
Please squash all commit into one @kunal-511

kunal-511 commented:
> Please squash all commit into one @kunal-511

Yes I will do this

xuelichao commented on src/portal/src/app/base/left-side-nav/projects/create-project/create-project.component.ts:
```diff
@@ -290,18 +308,53 @@ export class CreateProjectComponent
+            (value <= 0 && value !== -1)
+        ) {
+            this.bandwidthError =
+                'Please enter -1 or an integer  greater than 0.';
```
Remove extra white space between word 'integer' and 'greater'

xuelichao commented on src/portal/src/i18n/lang/zh-tw-lang.json:
```diff
@@ -245,12 +245,15 @@
     "STORAGE_QUOTA_TIP": "儲存配額上限只接受整數，上限為 '1024TB'。輸入 '-1' 表示無限配額。",
     "QUOTA_UNLIMIT_TIP": "專案所能使用的最大邏輯空間。輸入 '-1' 表示無限制配額。",
+    "BANDWIDTH": "Bandwidth",
+        "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
```
remove the extra indent

xuelichao commented on src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html:
```diff
@@ -16,6 +16,42 @@
+                >
+                <div class="flex">
+                    <input
+                        class="clr-input width-164 mr-10"
```
why does the [(ngModel)] not configured for the input element?

xuelichao commented on src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html:
```diff
@@ -16,6 +16,42 @@
+                    <div class="clr-select-wrapper mr-10">
+                        <select name="" id="">
+                            <option value="kbps" selected>kbps</option>
+                            <option value="Mbps">Mbps</option>
```
[(ngModel)] is not configured for select. why?

kunal-511 commented:
@stonezdj I have created a new PR with single commit and also addressed above issues. Please have a look [#20946](https://github.com/goharbor/harbor/pull/20946)
